test/common/network/address_impl_test.cc

```diff
@@ -336,6 +336,29 @@ TEST(PipeInstanceTest, BadAddress) {
                           "exceeds maximum UNIX domain socket path size");
 }
 
+// Validate that embedded nulls in abstract socket addresses are included and represented with '@'.
+TEST(PipeInstanceTest, EmbeddedNullAbstractNamespace) {
+#if defined(__linux__)
+  std::string embedded_null("@/foo/bar");
+  embedded_null[5] = '\0'; // Set embedded null.
+  PipeInstance address(embedded_null);
+  EXPECT_EQ("@/foo@bar", address.asString());
+  EXPECT_EQ("@/foo@bar", address.asStringView());
+  EXPECT_EQ(Type::Pipe, address.type());
+  EXPECT_EQ(nullptr, address.ip());
+#else
+  EXPECT_THROW(PipeInstance address(embedded_null), EnvoyException);
```

Comment:
Just a drive-by: I don't think this compiles on macOS because the definition of `embedded_null` is within the `#if defined(__linux__)` above

Reply:
Thank you!! Just fixed that.